src/lib.rs

```diff
@@ -223,10 +223,23 @@ pub fn binary_response<T>(status_code: T, body: Vec<u8>) -> Response
     http::response::Builder::new()
         .status(status_code)
         .header(http::header::CONTENT_LENGTH, format!("{}", body.len()).as_str())
+        .header(http::header::CONTENT_TYPE, "application/octet-stream")
         .body(body)
         .unwrap()
 }
 
+/// Sends  `blob` with that status code.
+pub fn binary_response_with_context_type<T>(status_code: T, body: Vec<u8>, context_type: &str) -> Response
```

Comment:
You're passing in a raw `&str` as the content type. [`.header(…)`](https://docs.rs/http/0.2.1/http/response/struct.Builder.html#method.header) can take a `TryFrom` for `HeaderValue`, which supports `&str` and other things too. Can you change the argument type to that?

Reply:
Do you mean as follows?

```
/// Sends  `blob` with that status code and content type.
pub fn binary_response_with_content_type<T, V>(status_code: T, content_type: V, body: Vec<u8>) -> Response
    where
        http::StatusCode: TryFrom<T>,
        <http::StatusCode as TryFrom<T>>::Error: Into<http::Error>,
        http::HeaderValue: TryFrom<V>,
        <http::HeaderValue as TryFrom<V>>::Error: Into<http::Error>,
{
    http::response::Builder::new()
        .status(status_code)
        .header(http::header::CONTENT_LENGTH, format!("{}", body.len()).as_str())
        .header(http::header::CONTENT_TYPE, content_type)
        .body(body)
        .unwrap()
}
```

This code causes a compile error.

```
240 |         .header(http::header::CONTENT_LENGTH, format!("{}", body.len()).as_str())
    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected type parameter `V`, found `&str`
```

If it is what you want to do, do you have any idea to solve this?

Reply:
oh yeah there's somethng funny about type inference, i remember that